### PR TITLE
Introduce the `contains_key` functionality for the `MapView` and `ReentrantCollectionView`.

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

The functionality is a very natural fit for the recently introduced functionality in the context.

## Proposal

The following was done:
* Introduce the functionality in `MapView` and its siblings.
* Introduce the functionality in `ReentrantCollectionView`and its siblings.
* A small optimization in the `SetView` which already had it.

The functionality was not introduced in `CollectionView` which likely still has some issues.

## Test Plan

The tests were significantly expanded to reflect the new functionality.

## Release Plan

Nothing relevant.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-views/src/map_view.rs [hunk 1]
```diff
@@ -207,6 +207,37 @@ where
     pub fn extra(&self) -> &C::Extra {
         self.context.extra()
     }
+
+    /// Tests if a key is contained in the container.
```
Comment (afck) [inline linera-views/src/map_view.rs:211]: Or maybe like in the standard library: "Returns `true` if the map contains a value for the specified key."

File: linera-views/src/reentrant_collection_view.rs [hunk 2]
```diff
@@ -325,6 +325,34 @@ where
...
+    ///   assert!(view.contains_key(&[0, 1]).await.unwrap());
+    ///   assert!(!view.contains_key(&[0, 2]).await.unwrap());
+    /// # })
+    /// ```
+    pub async fn contains_key(&self, short_key: &[u8]) -> Result<bool, ViewError> {
+        let updates = self.updates.lock().await;
+        Ok(match updates.get(short_key) {
+            Some(entry) => match entry {
+                Update::Set(_view) => true,
+                _entry @ Update::Removed => false,
```
Comment (afck) [inline linera-views/src/reentrant_collection_view.rs:347]:
```suggestion
                Update::Removed => false,
```

File: linera-views/tests/map_view_tests.rs [hunk 3]
```diff
@@ -106,6 +106,9 @@ async fn run_map_view_mutability<R: RngCore + Clone>(rng: &mut R) {
                             .collect::<Vec<_>>()
                     }
                 };
+                let test_view = view.map.contains_key(&key).await.unwrap();
+                let test_map = new_state_map.contains_key(&key);
```
Comment (afck) [inline linera-views/tests/map_view_tests.rs:110]: (Or maybe `actual` and `expected`.)

**Review by afck — APPROVED** [review]

File: linera-views/src/reentrant_collection_view.rs (hunk 2 above)
Comment (MathieuDutSik) [reply linera-views/src/reentrant_collection_view.rs:347]: Ok. Thanks.

File: linera-views/tests/map_view_tests.rs (hunk 3 above)
Comment (MathieuDutSik) [reply linera-views/tests/map_view_tests.rs:110]: I'm afraid I have to disagree here and prefer the `test_view / test/map`.

**Pull request merged**